drivers/st/clk/stm32mp1_clk.c

```diff
@@ -0,0 +1,1611 @@
+/*
+ * Copyright (C) 2018, STMicroelectronics - All Rights Reserved
+ *
+ * SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
```

Comment:
This is a file shared with U-Boot, and under dual licensing. So I'd prefer to keep the dual licensing comment here as well.

Reply:
This "disjunctive" dual licensing is a bit problematic for us. The difficulty is we have to be sure these files are the same as the ones in U-Boot and that all contributions are made by STMicro using this disjunctive license (rather than GPLv2). It also makes it unclear which project is the "upstream" for these files. It would be cleaner for us if these dual-licensed files existed in a separate OSS project that both TF-A and U-Boot pulled down from when needed. Is this something you would consider?

Reply:
U-Boot is the reference, the files can be found there:
http://git.denx.de/?p=u-boot.git;a=blob;f=drivers/clk/clk_stm32mp1.c;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=drivers/ram/stm32mp1/stm32mp1_ddr.c;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=drivers/ram/stm32mp1/stm32mp1_ddr.h;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=drivers/ram/stm32mp1/stm32mp1_ddr_regs.h;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=drivers/ram/stm32mp1/stm32mp1_ram.c;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=include/dt-bindings/clock/stm32mp1-clks.h;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=include/dt-bindings/clock/stm32mp1-clksrc.h;hb=HEAD
http://git.denx.de/?p=u-boot.git;a=blob;f=include/dt-bindings/reset-controller/stm32mp1-resets.h;hb=HEAD


All contributions in U-Boot by ST or other contributors will be under dual licence.
So ST can import/export fixes from one project to another.

If that's OK for you, I'll push the changes for copyrights.

Reply:
Hi. Thanks for the U-Boot links but they seem to be broken to me (404 not found). Once we have valid links that we can check we should be able to take these files. After merging we will update our readme to explain how we are treating these files.

Reply:
Yes, I have had the same issue about the links earlier in the afternoon, but it seems OK now.

Reply:
The last link was not correct, it is:
http://git.denx.de/?p=u-boot.git;a=blob;f=include/dt-bindings/reset/stm32mp1-resets.h;hb=HEAD